Makefile.dep

```diff
@@ -629,6 +629,20 @@ ifneq (,$(filter spiffs,$(USEMODULE)))
     USEMODULE += mtd
 endif
 
+ifneq (,$(filter fatfs_vfs,$(USEMODULE)))
+    USEPKG += fatfs
+    USEMODULE += vfs
+
+    ifeq ($(BOARD),native)
+      USEMODULE += fatfs_diskio_native
+      FATFS_DISKIO_NATIVE_DEFAULT_FILE ?= \"riot_fatfs_disk.img\"
+    else
+      USEMODULE += fatfs_diskio_sdcard_spi
+      USEMODULE += auto_init_storage
+    endif
```

Comment:
> From a centralization vs. modularization standpoint it doesn't seem so contrary.

True, but IMO the boards Makefile.dep is not the right place to put this - despite the fact I'm also prefering a modularized way.
>specific to any file system IO, not just FatFS

sure, to clarify what I meant: ~~a restriction that relates directly to the FatFs diskio layer~~ -> a restriction that relates directly to my currently sketchy implementation of the FatFs diskio layer, that should be using mtd in the future, as pointed out by @gebart.

>No, but they are the only one that can say specifically for the board "Hey, file systems go there".

Yes but no... I think the decision which device should actually be used by the FS should be made "as near as possible to the application". So for now (with the current shortcomings of the diskio layer) I'd like to keep this workaround there and have it removed from this single place once the diskio becomes almighty with the help of mtd ;)